_posts/en/newsletters/2020-09-23-newsletter.md

```diff
@@ -0,0 +1,293 @@
+---
+title: 'Bitcoin Optech Newsletter #116'
+permalink: /en/newsletters/2020/09/23/
+name: 2020-09-23-newsletter
+slug: 2020-09-23-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter describes a proposed soft fork to enable a new
+type of fee bumping and summarizes research into scripts that can never
+be spent because they require satisfying both timelocks and heightlocks.
+Also included are our regular sections with summaries of updates to
+services and client software, a list of new releases and release
+candidates, and changes to popular Bitcoin infrastructure software.
+
+## Action items
+
+*None this week.*
+
+## News
+
+- **Transaction fee sponsorship:** Jeremy Rubin [posted][rubin sponsor
+  bumping] a proposal to the Bitcoin-Dev mailing list for a soft fork to
+  add a new fee bumping mechanism that could be less vulnerable to abuse
+  than the current Replace-by-Fee ([RBF][topic rbf]) and
+  Child-Pays-For-Parent ([CPFP][topic cpfp]) methods.  [Transacting
+  pinning][topic transaction pinning] and other abuses
+  against the existing system can be used to attack contract
+  protocols.  These problems have led to partial solutions such as [anchor
+  outputs][topic anchor outputs] that work with specific protocols (e.g.
+  two-party LN channels) but aren't easy to generalize and so are
+  somewhat unsatisfactory.
+
+    At the consensus layer, Rubin proposes allowing transactions to
+    optionally contain a special final output that commits to the txids
+    of one or more other unconfirmed transactions.  The transaction with
+    the special output, called a *sponsor transaction*, may only be included in
+    a valid block if every one of the transactions it *sponsored* is
+    also included in that same block.  This means a miner who wants the
+    high feerate from a sponsor transaction will be incentivized to
+    confirm low-feerate sponsored transactions.  Beyond their special
+    sponsorship output, sponsor
+    transactions are normal Bitcoin transactions.
+
+    For the rules that control mempool acceptance and transaction relay
+    (the *policy layer*), Rubin suggests a set of simple changes that allow anyone to
+    sponsor any transaction currently in a mempool or to replace an
+    existing sponsor transaction with a higher feerate alternative that
+    makes the same commitment.  The goal is to ensure that, as long as a
+    sponsor transaction follows the rules and pays a high enough
+    feerate, it will propagate across the network without running afoul
+    of pinning or other attacks.
+
+    Rubin's proposal comes with a [reference implementation][rubin
+    refimpl] and discussion of design tradeoffs and forward
+    compatibility.  As of this writing, comments on the list have been
+    positive but non-committal---the proposal needs to not only provide
+    satisfactory solutions to existing problems but prove that it is
```

Comment:
Should we mention Suhas' concern about breaking a fundamental principle of bitcoin transactions? https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2020-September/018195.html

Reply:
Ugh, we should lobby for the mailing list to be turned off from noon on Tuesday until we publish.  :-)  I've added a quick description of Suhas's concern (preceded by Antoine's earlier-described concern that I also don't see an easy way to work around).

I think we would've also been ok with the previous text, which was open-ended about what would've happened next, so we can revert to that if the new text is controversial.

Reply:
In general, I don't think we should feel compelled to include any updates to the mailing list/PRs/etc that come in after the preceding weekend (ie when Dave starts writing the newsletter). The newsletter should strive to be a solid summary/analysis of what's happening, rather than trying to keep up with the very latest breaking news.

Reply:
@jnewbery 

> I don't think we should feel compelled to include any updates to the mailing list/PRs/etc that come in after the preceding weekend [...] The newsletter should strive to be a solid summary/analysis of what's happening, rather than trying to keep up with the very latest breaking news.

I agree.

Reply:
I agree too. I only mentioned this time since the newsletter had explicitly said "As of this writing, comments on the list have been positive" and yet Suhas had a pretty credible concern raised.

Reply:
@moneyball sure, but Suhas's concern wasn't raised until 47 hours after this PR was opened and several hours after all previous reviewer comments had been addressed.  :-)  That's why I like the phrase "as of this writing" (rather than "as of this publication"), since it references the state of affairs present during the early production of the newsletter, rather than any changes that occur during review.

In any case, I think it's absolutely fine for anyone to point out when our content has fallen behind later developments, and I'll try to make adjustments when I can, but I think @jnewbery's point and our general agreement means that we shouldn't feel obligated to make changes as long as we believe that what we wrote was accurate at the time the PR was opened.  A nice benefit of a weekly newsletter is that we're only ever seven days away from being able to update readers about the latest developments on a subject (and, if that's too long for them, they can subscribe to the same sources we use---which is a benefit of our insistence of only sourcing stories from public information).